### PR TITLE
Fix display of weather for selected cities

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2482

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/WeatherController.java [hunk 1]
```diff
@@ -50,6 +61,15 @@ public class WeatherController {
 	{
 		sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
 	}
+	
+	protected static final Map<String, Integer> SETTLEMENT_TYPES;
+	
+	static {
+		SETTLEMENT_TYPES = new HashMap<>();
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/WeatherController.java:68]: Use CityType structure, what this constant means? Approximate population could be part of CityType class

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/WeatherController.java [hunk 2]
```diff
@@ -109,27 +129,121 @@ public ResponseEntity<?> getWeatherForecast(@RequestParam double lat,
...
+				double lat2 = o2.location.getLatitude();
+				double lon2 = o2.location.getLongitude();
+				
+				double distance1 = Math.sqrt(Math.pow(centerLat - lat1, 2) + Math.pow(centerLon - lon1, 2));
+				double distance2 = Math.sqrt(Math.pow(centerLat - lat2, 2) + Math.pow(centerLon - lon2, 2));
+				
+				double rating1 = Math.log10(population1 + 1.0) - distance1;
+				double rating2 = Math.log10(population2 + 1.0) - distance2;
+				
+				return Double.compare(rating2, rating1);
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/WeatherController.java:240]: better to create method getRating(o1.object) so we don't need copy - ``` Double.compare(getRating(o1.object), getRating(o2.object));```

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexWeatherData.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class IndexWeatherData {
 	public static final int NEAREST_NEIGHBOOR_INTERPOLATION = 0;
 	public static final int BILINEAR_INTERPOLATION = 1;
 	public static final int BICUBIC_INTERPOLATION = 2;
-	public static int INTERPOLATION = BICUBIC_INTERPOLATION;
+	public static int INTERPOLATION = BILINEAR_INTERPOLATION;
```
Comment (vshcherb) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexWeatherData.java:23]: Leave comment that bicubic couldn't be used for clouds (%), precipitation

**Pull request merged**